raiden/raiden_service.py

```diff
@@ -240,11 +236,6 @@ def __init__(
         self.wal: Optional[wal.WriteAheadLog] = None
         self.snapshot_group = 0
 
-        # This flag will be used to prevent the service from processing
-        # state changes events until we know that pending transactions
-        # have been dispatched.
-        self.dispatch_events_lock = Semaphore(1)
```

Comment:
This doesn't have to be a lock, changed it to a flag and add a proper documentation